process/dataValidators/txValidator.go

```diff
@@ -115,6 +121,12 @@ func (txv *txValidator) getSenderUserAccount(
 }
 
 func (txv *txValidator) checkBalance(interceptedTx process.InterceptedTransactionHandler, account state.UserAccountHandler) error {
+	rTx, ok := interceptedTx.Transaction().(relayedV3TransactionHandler)
+	if ok && len(rTx.GetRelayerAddr()) > 0 {
+		// early return if this is a user tx of relayed v3, no need to check balance
```

Comment:
check balance of relayer is a must.

Like the transaction sender is the relayer, if it has InnerTransaction - the inner TX should be for the user. First we execute the balance transfer from the relayer to the user, and execute the inner TX after. 

GetRelayerAddr should be a field in the InnerTransaction and in the SmartContractResults.

Reply:
it won't apply for the relayed transaction

this check is only valid for the inner transaction. Reason: only transaction is castable to this structure, then only an inner transaction would have the relayer address set

the reason why this bypass is needed is because now, at api level, we also [create the inner transaction](https://github.com/multiversx/mx-chain-go/blob/relayedV3/api/groups/transactionGroup.go#L310), which would fail in the following scenario:
- User A with 0 egld signs a tx to move some ESDT
- Relayer B will pay the fee for it
- Since A does not have EGLD, the inner tx creation at API level would have failed

Reply:
Ok. I get that this is for user tx only. so it is the check for inner tx

Reply:
exactly, but does not work as expected.. on it

Reply:
completely removed it, not needed